internal/db/db.go

```diff
@@ -262,9 +262,41 @@ func (db *db) AddDocActorRelationship(
 		return client.AddDocActorRelationshipResult{}, err
 	}
 
+	err = db.publishDocUpdateEvent(ctx, docID, collection)
+	if err != nil {
+		return client.AddDocActorRelationshipResult{}, err
+	}
+
 	return client.AddDocActorRelationshipResult{ExistedAlready: exists}, nil
 }
 
+func (db *db) publishDocUpdateEvent(ctx context.Context, docID string, collection client.Collection) error {
+	headsIterator, err := NewHeadBlocksIterator(ctx, db.multistore.Headstore(), db.Blockstore(), docID)
+	if err != nil {
+		return err
+	}
+	defer headsIterator.Close()
+
+	for {
+		hasValue, err := headsIterator.Next()
+		if err != nil {
+			return err
+		}
+		if !hasValue {
+			break
+		}
+
+		updateEvent := event.Update{
+			DocID:      docID,
+			Cid:        headsIterator.CurrentCid(),
+			SchemaRoot: collection.Schema().Root,
+			Block:      headsIterator.CurrentRawBlock(),
+		}
+		db.events.Publish(event.NewMessage(event.UpdateName, updateEvent))
+	}
+	return nil
+}
+
 func (db *db) DeleteDocActorRelationship(
```

Comment:
todo: Don't we need to do the exact same thing for when we delete the relationship. i.e. `DeleteDocActorRelationship`

Reply:
the update event upon granting access is just to notify other nodes that they can try fetching encryption keys again.
I'm not sure why you would do it here. I can't even think of a test case.

Reply:
> Maybe add a test, that after a document is shared its access is revoked, and then again it's access is reshared again. In this case keys already exist and it should still be accessible again.

I just want to point out that revoking access is only possible for future versions of a document. Not for versions that have already been shared.

Reply:
Yes indeee